backend/services/implementations/prerequisiteService.ts

```
@@ -0,0 +1,131 @@
+      throw error;
+    }
+  }
+
+  async createPrerequisite(
+    prerequisite: PrerequisiteRequestDTO,
+  ): Promise<PrerequisiteResponseDTO> {
+    let newPrerequisite: Prerequisite | null;
+    try {
+      newPrerequisite = await prisma.prerequisite.create({
+        data: {
+          name: prerequisite.name,
+          completed: prerequisite.completed,
+          requiresAdminVerification: prerequisite.requiresAdminVerification,
+        },
+      });
+    } catch (error) {
+      Logger.error(`Failed to create prerequisite. Reason = ${error.message}`);
+      throw error;
+    }
+    return {
+      id: String(newPrerequisite.id),
+      name: newPrerequisite.name,
+      completed: newPrerequisite.completed,
+      requiresAdminVerification: newPrerequisite.requiresAdminVerification,
+    };
+  }
+
+  async updatePrerequisite(
+    prerequisiteId: string,
+    prerequisite: PrerequisiteRequestDTO,
+  ): Promise<PrerequisiteResponseDTO | null> {
+    let updateResult: Prerequisite | null;
+    try {
+      updateResult = await prisma.prerequisite.update({
+        where: { id: Number(prerequisiteId) },
+        data: {
+          name: prerequisite.name,
+          completed: prerequisite.completed,
+          requiresAdminVerification: prerequisite.requiresAdminVerification,
+        },
+      });
+
+      if (!updateResult) {
+        throw new Error(`Prerequisite id ${prerequisiteId} not found`);
+      }
+    } catch (error) {
+      Logger.error(`Failed to update prerequisite. Reason = ${error.message}`);
+      throw error;
+    }
+    return {
+      id: String(updateResult.id),
+      name: updateResult.name,
+      completed: updateResult.completed,
+      requiresAdminVerification: updateResult.requiresAdminVerification,
+    };
+  }
+
+  async deletePrerequisite(prerequisiteId: string): Promise<void> {
+    try {
+      const prerequisiteToDelete = await prisma.prerequisite.findUnique({
```

Is this needed? I think the `delete` call should return an error if it can't find the prerequisite

---

Good call! I've removed that line and the associated checks